### PR TITLE
NcpBase: Fix the model for adding/removing IPv6 addresses.

### PR DESCRIPTION
**Pull request by abtink**
This change addresses issue #204 by adding an array of elements of type `otNetifAddress` to `NcpBase` class. The array elements are then used for adding/removing of Ipv6 addresses.

### CODE REVIEW

**Comment by abtink** [comment]
@jwhui , @darconeous can you please check this?

File: src/ncp/ncp_base.cpp [hunk 1]
```diff
@@ -2486,37 +2488,60 @@ void NcpBase::InsertPropertyHandler_IPV6_ADDRESS_TABLE(uint8_t header, spinel_pr
...
+                       &valid_lifetime
                    );
 
     if (parsedLength > 0)
     {
-        netif_addr.mAddress = *addr_ptr;
-        errorCode = otAddUnicastAddress(&netif_addr);
+        for (unsigned i = 0; i < sizeof(mNetifAddresses) / sizeof(mNetifAddresses[0]); i++)
+        {
+            if (mNetifAddresses[i].mNext == NULL)
```
Comment (jwhui) [inline src/ncp/ncp_base.cpp]: Couple comments:
1. I don't think checking `mNext` is robust (e.g. in the case that the entry happens to be at the tail).  The most reliable way with the current API is to use `otGetUnicastAddresses` and walk the list.
2. Think about handling the case of adding an address that has been added before.  Does that return an error?  Does that simply update `mPrefixLength`, `mPreferredLifetime`, and/or `mValidLifetime`?

Comment (abtink) [inline src/ncp/ncp_base.cpp]: About (1) yes absolutely (was a silly mistake by me to use the next)...
about 2) I think  we should update the prefix len and lifetimes...

Comment (jwhui) [inline src/ncp/ncp_base.cpp]: For (1), you could use the trick as you described.

For (2), I think it's fine to succeed and update the parameters.

**Comment by abtink** [comment]
Modified the logic for adding a previously added address: In such as case, we now update the prefix len and the address lifetime values and consider the operation success.

File: src/ncp/ncp_base.hpp [hunk 2]
```diff
@@ -285,6 +285,11 @@ class NcpBase
                                                       uint16_t value_len);
 
 private:
+    enum
+    {
+        kNetifAddressListSize = 8,
+        kUnusedNetifAddressPrefixLen = 0xff,   // Prefix len value to indicate an unused/available NetifAddress element in mNetifAddresses.
```
Comment (jwhui) [inline src/ncp/ncp_base.hpp]: nit: can we keep the line width to 120 characters max?

**Comment by jwhui** [comment]
LGTM, @darconeous?

**Comment by darconeous** [comment]
👍 

Good job

**Pull request merged**